src/test/java/bio/terra/service/resourcemanagement/google/GoogleProjectServiceTest.java

```diff
@@ -85,4 +86,20 @@ public void testAppEngineOpIdExtraction() {
         .isInstanceOf(AppengineException.class)
         .hasMessageStartingWith("Operation Name does not look as expected");
   }
+
+  @Test
+  public void testProjectLabelClean() {
+    String tooLongName = "workflow_launcher_testing_dataset5243fe12db16406789e76e98dcf3aebd";
+    assertEquals("Project label original length should be 65", tooLongName.length(), 65);
+    String trimmedName = GoogleProjectService.cleanForLabels(tooLongName);
+    assertEquals("Project label should be trimmed down when too long", trimmedName.length(), 63);
+
+    String nameWithCharacters = "workflow!_launcher+TESTING_dataset5243fe12db1640";
+    String expectedCleanedName = "workflow-_launcher-testing_dataset5243fe12db1640";
+    String cleanedName = GoogleProjectService.cleanForLabels(nameWithCharacters);
```

Comment:
You could test both cases at once using a longer name for the second assert, to test truncation and character replacement.

Reply:
This is true - I was thinking I wanted to keep them separate so it would be clear which cleaning step failed.